app/gui2/src/components/GraphEditor/widgets/WidgetVector.vue

```diff
@@ -45,16 +48,11 @@ const navigator = injectGraphNavigator(true)
 <script lang="ts">
 export const widgetDefinition = defineWidget(WidgetInput.isAstOrPlaceholder, {
   priority: 500,
-  score: (props) => {
-    if (props.input.dynamicConfig?.kind === 'Vector_Editor') return Score.Perfect
-    else if (props.input.expectedType?.startsWith('Standard.Base.Data.Vector.Vector'))
-      return Score.Good
-    else if (props.input.value instanceof Ast.Ast) {
-      return props.input.value.children().next().value.code() === '[' ?
-          Score.Perfect
-        : Score.Mismatch
-    } else return Score.Mismatch
-  },
+  score: (props) =>
+    props.input.dynamicConfig?.kind === 'Vector_Editor' ? Score.Perfect
+    : props.input.expectedType?.startsWith('Standard.Base.Data.Vector.Vector') ? Score.Good
+    : props.input.value instanceof Ast.Vector ? Score.Perfect
```

Comment:
nit: I suggest switching these lines. If we have vector Ast *and* the expected type is Vector, it should be a perfect match, not `Good`.